servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -298,7 +298,10 @@ public FlushStrategy defaultFlushStrategy() {
 
         private Completable handleRequestAndWriteResponse(final Single<StreamingHttpRequest> requestSingle,
                                                           final boolean handleMultipleRequests) {
-            final Completable exchange = requestSingle.flatMapCompletable(rawRequest -> {
+            final Completable exchange = requestSingle.whenOnError(t -> {
+                // If an error happened before we could start writing, we need to close the channel.
+                nettyChannel().close();
```

Comment:
when we close under the hood it can be difficult for users to understand who initiated closure. Consider either:
- moving this close operation into `ErrorLoggingHttpSubscriber` after it logs the reason
- adding a comment that `ErrorLoggingHttpSubscriber` will log the exception if required.

Reply:
Good point, I've moved closure to `ErrorLoggingHttpSubscriber`